utils/src/cmd.c

```diff
@@ -42,22 +45,35 @@ static struct argp_command *find_command(char *name)
 	return NULL;
 }
 
-static void usage(void)
+static void print_cmds_for_group(int group)
 {
 	struct argp_command *com;
 	int largest = 0;
 
-	fprintf(stderr, "usage: scoutfs <command> [<args>]\n"
-	       "Commands:\n");
-
+	/* Base alignment on all groups */
 	cmd_for_each(com)
 		largest = max(strlen(com->name), largest);
 
 	cmd_for_each(com) {
-		fprintf(stderr, "  %*s %s\n  %*s %s\n",
-			largest, com->name, com->argp->args_doc,
-			largest, "", com->argp->doc);
+		if (com->group == group) {
+			fprintf(stderr, "  %*s %s\n  %*s %s\n",
+				largest, com->name, com->argp->args_doc,
+				largest, "", com->argp->doc);
+		}
 	}
+
+}
+
+static void usage(void)
+{
+	fprintf(stderr, "usage: scoutfs <command> [<args>]\n");
+
+	fprintf(stderr, "\nBase commands:\n");
```

Comment:
I'm wondering if we really need these groups.  I can't find a concise set of groups that makes much sense, I feel like there's too many ways to slice the commands (modifying vs read-only?), and there's so few commands to begin with.  Should we just sort them and be done with it?

Reply:
Sort them how?

The groups are really me trying to sort by popularity, or "likelihood that the user will ever use" a subcommand. Let's call this "importance". I think putting important subcommands first (so the user sees them first) has value. Should I just do this and not try to group them?

Reply:
I was thinking just sorting the commands by name in a single list.  But I'll be less of a stick in the mud and trust that the groupings have value.  I came up with:

```Core Administration:
        df, mkfs.
Additional Information:
        counters, stat, statfs
Search Acceleration:
        ino-path, search-xattrs, walk-inodes
Archival Agent Support:
        data-waiting, data-wait-err, listxattr-hidden, setattr,
        stage, release 
Debugging:
        print
```

Reply:
cool, thanks, these sound fine. btw just to mention I'm assuming the "Core Administration" group will add a "tune" subcommand at some point.